sphinxcontrib/images.py

```diff
@@ -157,17 +157,16 @@ def run(self):
         if self.is_remote(self.arguments[0]):
             img['remote'] = True
             if download:
-                img['uri'] = os.path.join('_images', hashlib.sha1(self.arguments[0].encode()).hexdigest())
+                reluri = os.path.join(conf['cache_path'], hashlib.sha1(self.arguments[0].encode()).hexdigest())
+                img['uri'] = os.path.join('/', reluri) # relative to source dir
```

Comment:
I'm a little confused about this, do you need to `os.path.join` the relative path to `/`? Is this actually the URL that Sphinx will use to reference the image in the HTML? If it's actually that it might make sense to use `urllib.parse.urljoin` in order to make it more obvious what you're doing.

```python
# *nix
>>> os.path.join('/', 'test/key')
'/test/key'
>>> urllib.parse.urljoin('/', 'test/key')
'/test/key'
```

Reply:
I think I wrote a confusing comment :'(.

The prefixed `/` means the uri is "relative to source dir"(rootof sphinx project) but not "relative to current document's dir".  Without it, sphinx will try load image from `srcdir/foo/bar/_images`, while our `cache_path` is `srcdir/_images/`.

And the URI is actualy  a filesystem path here, so I think we need ``os.path.join``.

Reply:
Why wasn't this an issue before this PR? Thumbnail directives with `:download: true` were copied correctly to `_build/html/_images/` and the URL set correctly before this PR (e.g. in 0.9.3).

Images (local or remote) in `index.html` have URLs `_images/<imagefile>` and images in `subdir/subfile.rst` have URLs `../_images/<imagefile>` (because the html-file resides in a subdirectory compared to index.html `../_images` are used).